channelserver/src/meta.rs

```diff
@@ -142,6 +145,84 @@ fn get_ua(headers: &http::HeaderMap, log: Option<&Addr<logging::MozLogger>>) ->
     None
 }
 
+fn check_address(proxy_list:&[IpNet], host:&str) -> Option<String> {
+    /// Return if an address is NOT part of the allow list
+    let test_addr: IpAddr = match host.parse() {
+        Ok(a) => a,
+        Err(err) => {
+            return None
+        }
+    };
+    for proxy_range in proxy_list {
+        if ! proxy_range.contains(&test_addr) {
+            return Some(host.to_owned())
+        }
+    }
+    None
+}
+
+fn get_remote(headers: &http::HeaderMap, proxy_list: &[IpNet]) -> Option<String> {
+    // Actix determines the connection_info.remote() from the first entry in the
+    // Forwarded then X-Fowarded-For then peer name. The problem is that any
+    // of those could be multiple entries or may point to a known proxy.
+    //
+    // Check the [FORWARDED](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) 
+    // header first.
+    // Forwarded is a comma separated set of sub-fields that indicate the prior connection. The
+    // `for=` sub key identifies the origin. Each new server should be appended to 
+    // the end of this list. 
+    // (e.g. 
+    // `Forwarded: by=<identifier>; for=<identifier>; host=<host>; proto=<http|https>, for=<identifier>...` 
+    // ) 
```

Comment:
It's still not clear to me why we want to check the `Forwarded` header here.  Checking it makes sense if we trust the client and all intermediate proxies to correctly report the originating IP address of the request.  But since we want to use that IP for security measures, we can't trust anything outside the boundaries of our own infrastructure.

The question we want to answer here is: given this HTTP request, received from this client IP address, against what IP address should we apply our rate-limiting and geo-lookup security measures?

If this rust server was directly facing the internet, the answer is "the client IP from which we received the request", because there's no other trusted infrastructure in the equation.  The user might have been behind some well-behaved proxies, but we can't trust that, and we have to apply security measures based on the IP address of the proxy.

If this rust server is behind a load-balancing proxy with known IP address, then the answer is:
* If the client IP is not a trusted proxy, then use the client IP itself.
* Otherwise, use the last item from `X-Forwarded-For` that is not a trusted proxy.
We trust that the proxy has correctly appended the client's real IP address to the end of the `X-Forwarded-For` list.

I can't think of a scenario in which we want to trust data from the `Forwarded` header; AFAICT it's not supported by ELBs or ALBs, at least by default.

Reply:
Sorry, I got confused. Going to pull `Forward` handling entirely.